components/movementsensor/adxl345/adxl345.go

```diff
@@ -248,10 +251,12 @@ func NewAdxl345(
 			case <-timer.C:
 				// The registers with data are 0x32 through 0x37: two bytes each for X, Y, and Z.
 				rawData, err := sensor.readBlock(sensor.cancelContext, 0x32, 6)
+				// Record the errors no matter what: if the error is nil, that's useful information
+				// that will prevent errors from being returned later.
+				sensor.mu.Lock()
+				sensor.err.Set(err)
```

Comment:
[q] Do all the sensors need to have similar error-setting patterns? Or is this mutex lock/unlcok particular because of something in the i2c, rather than just calling `Set` as other sensors do?

Reply:
Good suggestion! I had just moved the code without thinking about it, but `sensor.err` contains its own mutex and doesn't need this any more. This is a holdover from back when `sensor.err` was a raw `error` value.